### ISSUE TITLE
Stubbing options returned by a composed generator using helpers.createDummyGenerator?

### ISSUE THREAD

**Issue by willprice** [yeoman/generator#910]
I've got a generator that I'm composing with generator-license, I'd like to access the license set by the user in my generator so that I can inject it into the README I'm generating in my generator, in testing I'm stubbing the license generator using `helpers.createDummyGenerator()`, but I can't seem to find any documentation or flex points from the source that will allow me to inject the options that the fake generator should return.

**Comment by SBoudrias** [comment]
Composed generator don't return anything. What do you need the dummy generator to do?

**Comment by willprice** [comment]
Hi SBoudrias, sorry I didn't articulate myself very well. Let me try again.

I'll explain what I'm trying to do, how I've done it, and how I'm struggling to write a test for it. Perhaps you can suggest an alternate approach.

## Desired functionality
I want to create a generator the produces a skeleton of an Ansible role or playbook, for each of these I'd like the user to select a license, once they've selected a license I'd like to create a LICENSE file and also insert the chosen license in the README and also in `meta/main.yml` if the user is generating a role.

## Current implementation
Instead of writing my own generator to produce a license, I'd like to reuse the existing generator: `generator-license` which covers my use case well, I'd like to get a handle on the license the user has selected in my generator so that I can use it when I'm templating some files. Currently I'm composing my generator with `generator-license` like so:
```js
var licenseGeneratorAnswers = {};

module.exports = generators.Base.extend({
   constructor: function () {
     generators.Base.apply(this, arguments)
     this.composeWith('license', licenseGeneratorAnswers, { link: 'weak' })
   }
   template: function () {
     this.fs.copyTpl(this.templatePath('README.md.ejs'), this.destinationPath('README.md'), {
        license: licenseGeneratorAnswers.license,
        ...
   }
}
```

## Testing?
I'd like to test my code, so I'd like to either run the license generator by setting the prompts it asks for, or stubbing the generator and faking the answers it returns, but I can't seem to figure out how to do either of these.


Does that make more sense? Let me know if you need me to clarify anything.

**Comment by SBoudrias** [comment]
There's not really an easy way to do this ATM and I don't think that'll change really soon.

Basically, moving data from the bottom-up is kind of hard and really error prone. A way this could be done is to read the value generator-license write to the `package.json` file as `license`. For this, you could launch the generator requiring the value inside the `end` priority to make sure generator-license runs before.

Also, there's the option to extract the prompt logic to its own module - like we did with https://github.com/SBoudrias/inquirer-npm-name inside https://github.com/yeoman/generator-node/blob/master/generators/app/index.js#L120

I know this is currently a pain, and I've been considering ways to make this easier for our [next major API iteration](https://github.com/yeoman/generator/issues/894) - that being said, this new API will probably not be release soon.

I'll close this issue given we're planning to ease the pain in the future and provided some potential alternatives for now.
